posthog/heatmaps/heatmaps_api.py

```diff
@@ -282,3 +293,248 @@ def _return_scroll_depth_response(query_response: HogQLQueryResponse) -> respons
+
+
+# Heatmap Screenshot functionality
+
+
+class HeatmapScreenshotResponseSerializer(serializers.ModelSerializer):
+    created_by = UserBasicSerializer(read_only=True)
+    snapshots = serializers.SerializerMethodField()
+
+    class Meta:
+        model = SavedHeatmap
+        fields = [
+            "id",
+            "short_id",
+            "name",
+            "url",
+            "data_url",
+            "target_widths",
+            "type",
+            "status",
+            "has_content",
+            "snapshots",
+            "deleted",
+            "created_by",
+            "created_at",
+            "updated_at",
+            "exception",
+        ]
+        read_only_fields = [
+            "id",
+            "short_id",
+            "status",
+            "has_content",
+            "created_by",
+            "created_at",
+            "updated_at",
+            "exception",
+        ]
+
+    def get_snapshots(self, obj: SavedHeatmap) -> list[dict]:
+        # Expose metadata of generated snapshots (width + readiness)
+        snaps = []
+        for snap in obj.snapshots.all():
+            snaps.append(
+                {
+                    "width": snap.width,
+                    "has_content": bool(snap.content or snap.content_location),
+                }
+            )
+        return sorted(snaps, key=lambda s: s["width"]) if snaps else []
+
+
+class HeatmapScreenshotViewSet(TeamAndOrgViewSetMixin, viewsets.GenericViewSet):
+    scope_object = "INTERNAL"
+    throttle_classes = [ClickHouseBurstRateThrottle, ClickHouseSustainedRateThrottle]
+    serializer_class = HeatmapScreenshotResponseSerializer
+    authentication_classes = [TemporaryTokenAuthentication]
+    queryset = SavedHeatmap.objects.all()
+
+    def safely_get_queryset(self, queryset):
+        return queryset.filter(team=self.team)
+
+    @action(methods=["GET"], detail=True)
+    def content(self, request: request.Request, *args: Any, **kwargs: Any) -> HttpResponse:
+        screenshot = self.get_object()
+        if screenshot.deleted:
+            return response.Response(status=status.HTTP_404_NOT_FOUND)
+
+        # Pick requested width or default
+        try:
+            requested_width = int(request.query_params.get("width", 1024))
+        except Exception:
+            requested_width = 1024
+
+        # Try exact match snapshot
+        snapshot = screenshot.snapshots.filter(width=requested_width).first()
+
+        # If not found, pick closest by absolute difference among available snapshots
+        if not snapshot:
+            all_snaps = list(screenshot.snapshots.all())
+            if all_snaps:
+                snapshot = min(all_snaps, key=lambda s: abs(s.width - requested_width))
+
+        if not snapshot:
+            # Nothing generated yet
+            response_serializer = HeatmapScreenshotResponseSerializer(screenshot)
+            return response.Response(response_serializer.data, status=status.HTTP_202_ACCEPTED)
```

Comment:
HTTP 202 is explicitly a successful response - I think the semantics of that response would be a better fit here if this endpoint would kick off an async task to actually generate the screenshot.

Since it doesnt I think HTTP 404 is a better response here....